go/base_client.go

```diff
@@ -1211,13 +1211,13 @@ func (client *baseClient) Append(ctx context.Context, key string, value string)
 //	An empty string is returned if the keys do not exist or have no common subsequences.
 //
 // [valkey.io]: https://valkey.io/commands/lcs/
-func (client *baseClient) LCS(ctx context.Context, key1 string, key2 string) (string, error) {
+func (client *baseClient) LCS(ctx context.Context, key1 string, key2 string) (*models.LCSMatch, error) {
```

Comment:
Somewhere we return a ptr, somewhere an object. What is better?
Need to align APIs.

Reply:
When we are returning a struct we should probably be returning a pointer, unless the struct is dead simple. Returning by value involves cloning the object which has performance and memory usage implications.

Reply:
Should we update our other APIs to avoid misaligment?